drivers/input/input_renesas_ra_ctsu.c

```diff
@@ -0,0 +1,676 @@
+		.p_button_status = &CTSU_GROUP_VAR_NAME(id, _button_data),                         \
+		.p_slider_position = CTSU_GROUP_VAR_NAME(id, _slider_data),                        \
+		.p_wheel_position = CTSU_GROUP_VAR_NAME(id, _wheel_data),                          \
+		.p_button_cb = CTSU_GROUP_VAR_NAME(id, _button_cb),                                \
+		.p_slider_cb = CTSU_GROUP_VAR_NAME(id, _slider_cb),                                \
+		.p_wheel_cb = CTSU_GROUP_VAR_NAME(id, _wheel_cb),                                  \
+	};                                                                                         \
+                                                                                                   \
+	DEVICE_DT_DEFINE(id, renesas_ra_ctsu_group_init, NULL, &CTSU_GROUP_VAR_NAME(id, _data),    \
+			 &CTSU_GROUP_VAR_NAME(id, _cfg), POST_KERNEL, CONFIG_INPUT_INIT_PRIORITY,  \
+			 NULL);
+#endif
+
+static int renesas_ra_ctsu_init(const struct device *dev)
+{
+	const struct renesas_ra_ctsu_cfg *ctsu_cfg = dev->config;
+	struct renesas_ra_ctsu_data *data = dev->data;
+	int ret;
+
+	if (!device_is_ready(ctsu_cfg->clock)) {
+		return -ENODEV;
+	}
+
+	/* Perform discharge process for the TSCAP pin */
+	ret = gpio_pin_configure_dt(&ctsu_cfg->tscap_pin, GPIO_OUTPUT_LOW);
+	if (ret != 0) {
+		return ret;
+	}
+
+	/* Wait 10 usec for discharge to complete before switching to the CTSU pin function */
+	k_busy_wait(10);
+
+	ret = pinctrl_apply_state(ctsu_cfg->pcfg, PINCTRL_STATE_DEFAULT);
+	if (ret != 0) {
+		return ret;
+	}
+
+	ret = clock_control_on(ctsu_cfg->clock, (clock_control_subsys_t)&ctsu_cfg->clock_subsys);
+	if (ret != 0) {
+		return ret;
+	}
+
+	k_sem_init(&data->scanning, 0, 1);
+	k_queue_init(&data->scan_q);
+
+	ctsu_cfg->make_thread(dev);
+	ctsu_cfg->irq_config();
+
+	return 0;
+}
+
+#define CTSU_DRV_THREAD_PRIORITY   CONFIG_INPUT_RENESAS_RA_CTSU_DRV_PRIORITY
+#define CTSU_DRV_THREAD_STACK_SIZE CONFIG_INPUT_RENESAS_RA_CTSU_DRV_STACK_SIZE
+
+#define RENESAS_RA_CTSU_DEFINE(inst)                                                               \
+	PINCTRL_DT_INST_DEFINE(inst);                                                              \
+	K_THREAD_STACK_DEFINE(renesas_ra_ctsu_drv_stack##inst, CTSU_DRV_THREAD_STACK_SIZE);        \
+                                                                                                   \
+	static void renesas_ra_ctsu_thread##inst(void *dev, void *arg1, void *arg2)                \
+	{                                                                                          \
+		ARG_UNUSED(arg1);                                                                  \
+		ARG_UNUSED(arg2);                                                                  \
+		renesas_ra_ctsu_drv_handler(dev);                                                  \
+	}                                                                                          \
```

Comment:
why not just use `renesas_ra_ctsu_drv_handler` as argument for `k_thread_create`? there's no references to `inst` in the body